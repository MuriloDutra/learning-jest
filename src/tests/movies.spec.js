import movies from "../resources/movies";

describe("Favorite Movies", () => {
  let myMovies = [];
  beforeEach(() => {
    myMovies = [
      {
        title: "Age of Ultron",
        rate: null,
      },
    ];
  });

  test("Can add a movie", () => {
    movies.add(myMovies, "Django");
    expect(myMovies).toMatchSnapshot();
  });

  test("Rate a movie", () => {
    movies.rate(myMovies[0], 5);
    expect(myMovies).toMatchSnapshot();
  });
});
