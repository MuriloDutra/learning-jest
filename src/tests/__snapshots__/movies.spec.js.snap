// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Movies Can add a movie 1`] = `
[
  {
    "rate": null,
    "title": "Age of Ultron",
  },
  {
    "rate": null,
    "title": "Django",
  },
]
`;

exports[`Favorite Movies Rate a movie 1`] = `
[
  {
    "rate": 5,
    "title": "Age of Ultron",
  },
]
`;
